java/dataset/src/main/cpp/jni_wrapper.cc

```diff
@@ -261,6 +264,26 @@ void JNI_OnUnload(JavaVM* vm, void* reserved) {
   default_memory_pool_id = -1L;
 }
 
+/// Iterate over an object array of Tables Name on position `i` and theirs
+/// Memory Address representation on `i+1` position linearly.
+/// Return a mapping of the Table Name to Query as a Key and a RecordBatchReader
+/// as a Value.
+std::unordered_map<std::string, std::shared_ptr<arrow::Table>> ToMapTableToArrowReader(JNIEnv* env, jobjectArray& str_array) {
+  std::unordered_map<std::string, std::shared_ptr<arrow::Table>> map_table_to_record_batch_reader;
+  int length = env->GetArrayLength(str_array);
+  std::shared_ptr<arrow::Table> output_table;
+  for (int pos = 0; pos < length; pos++) {
+    auto j_string_key = reinterpret_cast<jstring>(env->GetObjectArrayElement(str_array, pos));
+    pos++;
+    auto j_string_value = reinterpret_cast<jstring>(env->GetObjectArrayElement(str_array, pos));
+    auto* arrow_stream_in = reinterpret_cast<ArrowArrayStream*>(std::stol(JStringToCString(env, j_string_value)));
```

Comment:
We should catch errors from stol.

Reply:
This is internal evaluation, not exposed to the user to manipulate or setup that. Instead of that if needed try/catch this errors from stol?

Reply:
We should catch possible errors.

Reply:
Ok